src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1092,14 +1092,19 @@ namespace cryptonote
 
       results[i].blob_size = it->blob.size();
       results[i].weight = results[i].tx.pruned ? get_pruned_transaction_weight(results[i].tx) : get_transaction_weight(results[i].tx, it->blob.size());
-      ok &= add_new_tx(results[i].tx, results[i].hash, tx_blobs[i].blob, results[i].weight, tvc[i], tx_relay, relayed);
+      if (!add_new_tx(results[i].tx, results[i].hash, tx_blobs[i].blob, results[i].weight, tvc[i], tx_relay, relayed))
+      {
+        // Transactions with big tx_extra are still ok to be mined, so just ignore them and don't return false
+        if (!tvc[i].m_tx_extra_too_big)
+          ok = false;
```

Comment:
In short: the worst case, some connections that should've been dropped, will not be dropped sometimes, but otherwise invalid transactions will never be relayed or added to mempool.

Reply:
Please constructive notes only. This PR was created not to refactor the code.

Reply:
@UkoeHB I can change `tx_memory_pool::add_tx` to perform tx_extra check last, so second case will never happen. Then "tx extra is too big and something else" case will result in exactly the same node behavior as before this PR. Will it be good enough for you?

Reply:
I'm telling you it's not. tvc will record "too big tx extra", and it will still record `tvc.m_verifivation_failed = true;`, so all code handling tvc will treat this transaction as invalid. Converstaion over, I will not change this logic.

Reply:
Ok, this is first valid point I got in this conversation. It can potentially break `core_rpc_server::on_send_raw_tx` and `DaemonHandler::handleTxBlob`. I'll probably move the connection drop fix to `cryptonote_protocol_handler.inl` where it really belongs.